gadget/layout.go

```diff
@@ -169,13 +171,13 @@ func layoutVolumeStructures(volume *Volume, constraints LayoutConstraints) (stru
 
 		end := start + quantity.Offset(s.Size)
 		ps := LaidOutStructure{
-			VolumeStructure: &volume.Structure[idx],
+			GadgetStructure: &volume.Structure[idx],
```

Comment:
discussed that we probably want to make sure that matching fields between LaidOutStructure.OnDiskStructure and GadgetStructure should be set to match, except when it makes sense for them to diverge, in which case we need to be careful, for example for Size.

Also it might make sense to add methods like HasFilesytem, IsPartition, and a new Role() to LaidOutStructure